API Blueprint Specification.md

```diff
@@ -767,6 +776,16 @@ Defined by the `Request` keyword followed by an optional [identifier](#def-ident
 
     + Request <identifier> (<Media Type>)
 
+This section **should** include at least one of the following nested sections:
+
+- [`0-1` Headers section](#def-headers-section)
+- [`0-1` Attributes section](#def-attributes-section)
+- [`0-1` Body section](#def-body-section)
+- [`0-1` Schema section](#def-schema-section)
+- [`0-1` URI parameters section](#def-uriparameters-section)
+
+The URI parameters section included in this section describes the URI parameters attached to the request. They can be used to represent different responses based on different parameter values sent along with a request.
```

Comment:
> one overrides the other and there is no inheritance.

Why? Requests and responses can't have their own uriTemplate. So, they have to rely on the uriTemplate from the action (or above). So, why can't they inherit the parameters too?

I think they should definitely inherit the action parameters. We are adding the parameters in request to differentiate between different values of parameters which are inherited from the action.

An example would be:

```apib
### Search for a question [GET /questions{?query,limit}]

+ Parameters
    + query - Search query
    + limit: 20 (number, optional) - Maximum number of questions returned

+ Response 200 (application/json)

        [
            { "question": "Hello?" },
            { "question": "Favourite programming language?" },
            { "question": "API Description language preference" }
        ]

+ Request A search limited to one result
    + Parameters
        + query: language
        + limit: 1

+ Response 200 (application/json)

        [
            { "question": "Favourite programming language?" }
        ]
```

Here, `query` in action parameters is left empty and it's value will be filled by the request parameters.

"Request Parameters" object inherit "Action Parameters" object. No new members will be added since the uriTemplate will be the same. It's just going to be the change of values.